odoo/addons/base/models/res_country.py

```diff
@@ -79,16 +99,40 @@ class Country(models.Model):
         ('name_uniq', 'unique (name)',
             'The name of the country must be unique!'),
         ('code_uniq', 'unique (code)',
-            'The code of the country must be unique!')
+            'The code of the country must be unique!'),
+        ('iso_alpha3_uniq', 'unique (iso_alpha3)',
+            'The ISO 3166-1 alpha-3 of the country must be unique!'),
+        ('iso_num_uniq', 'unique (iso_num)',
+            'The ISO 3166-1 numeric of the country must be unique!'),
     ]
 
+    def _compute_code(self, fname, mapping):
+        for country in self:
+            country[fname] = mapping.get(country.code)
+
+    @api.depends('code')
+    def _compute_iso_alpha3(self):
+        self._compute_code('iso_alpha3', COUNTRY_CODE_2_TO_3)
+
+    @api.depends('code')
+    def _compute_iso_num(self):
+        self._compute_code('iso_num', COUNTRY_CODE_2_TO_NUM)
+
+    @api.depends('code')
+    def _compute_phone_code(self):
+        self._compute_code('phone_code', COUNTRY_CODE_2_TO_PHONE)
+
     def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
         if domain is None:
             domain = []
 
         ids = []
         if len(name) == 2:
-            ids = list(self._search([('code', 'ilike', name)] + domain, limit=limit, order=order))
+            ids = list(self._search([('code', '=', name.upper())] + domain, limit=limit, order=order))
+        elif len(name) == 3 and name.isalpha():
+            ids = list(self._search([('iso_alpha3', '=', name.upper())] + domain, limit=limit, order=order))
+        elif len(name) == 3 and name.isdecimal():
+            ids = list(self._search([('iso_num', '=', name)] + domain, limit=limit, order=order))
```

Comment:
Some numberic codes have 1 or 2 digits.  The condition `len(name) == 3` may be too strong IMO.

Reply:
> [ISO 3166-1 numeric codes are three-digit country codes defined in ISO 3166-1, part of the ISO 3166 standard published by the International Organization for Standardization (ISO)](https://en.wikipedia.org/wiki/ISO_3166-1_numeric)

Reply:
My bad, I misread the tuples and confused the ISO numeric code with the phone prefix.